src/main/kotlin/ar/edu/unahur/obj2/socios/Barrio.kt

```diff
@@ -0,0 +1,21 @@
+package ar.edu.unahur.obj2.socios
+
+abstract class Barrio {
+    abstract fun influirPropina(cliente: Cliente, propinaSegunAnimo : Int): Int
+}
+
+object LasRosas : Barrio() {
+    override fun influirPropina(cliente: Cliente, propinaSegunAnimo : Int) = 50
+}
+
+object LasLauchas : Barrio() {
+    override fun influirPropina(cliente: Cliente, propinaSegunAnimo : Int) = -propinaSegunAnimo / 2
+}
+
+object BarrioVerde : Barrio() {
+    override fun influirPropina(cliente: Cliente, propinaSegunAnimo : Int) = 200
+}
```

Comment:
Esto no es lo que se pedía...
> - **Barrio Verde** 💸: fija un mínimo de $200 para la propina.

Lo que significa que si el importe es menor a 200 tiene que devolver 200, de lo contrario el valor total del importe.
Una forma de resolverlo sería con `max()`
```suggestion
object BarrioVerde : Barrio() {
    override fun influirPropina(cliente: Cliente, propina : Int) = max(propina, 200)
}
```